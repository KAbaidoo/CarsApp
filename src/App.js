

import './App.css';
import { Toolbar, Typography , AppBar} from '@mui/material';
import Carlist from './components/Carlist';

function App() {
  return (
    <div className="App">
      <AppBar position="static">
        <Toolbar>
          <Typography variant='h6'>CarShop</Typography>
        </Toolbar>
      </AppBar>
      <Carlist />
    </div>
  );
}

export default App;
