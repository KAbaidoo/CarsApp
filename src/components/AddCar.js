import React, { useState } from "react";
import Dialog from "@mui/material/Dialog";
import DialogActions from "@mui/material/DialogActions";
import DialogContent from "@mui/material/DialogContent";
import DialogTitle from "@mui/material/DialogTitle";
import Button from "@mui/material/Button";
import TextField from "@mui/material/TextField";
import Stack from "@mui/material/Stack";

function AddCar(props) {
  const [open, setOpen] = useState(false);
  const [car, setCar] = useState({
    brand: "",
    model: "",
    color: "",
    manufactureYear: "",
    registerNumber: "",
    price: "",
  });

  function handleClickOpen() {
    setOpen(true);
  }

  function handleClose() {
    setOpen(false);
  }

  const handleChange = (event) => {
    setCar({ ...car, [event.target.name]: event.target.value });
  };

  const handleSave = () => {
    props.addCar(car);
    handleClose();
  };

  return (
    <div>
      <Button variant="contained" color="primary" onClick={handleClickOpen}>
        Add Car
      </Button>
      <Dialog open={open} onClose={handleClose}>
        <DialogTitle>Add New Car</DialogTitle>
        <DialogContent>
            <Stack spacing={2} mt={1}>
          <TextField
            autoFocus
            variant="standard"
            placeholder="Brand"
            onChange={handleChange}
            name="brand"
            value={car.brand}
          />
          <TextField
            variant="standard"
            placeholder="Model"
            onChange={handleChange}
            name="model"
            value={car.model}
          />
          <TextField
            variant="standard"
            placeholder="Color"
            onChange={handleChange}
            name="color"
            value={car.color}
          />
          <TextField
            variant="standard"
            placeholder="Year"
            onChange={handleChange}
            name="manufactureYear"
            value={car.manufactureYear}
          />
          <TextField
            variant="standard"
            placeholder="Register Number"
            onChange={handleChange}
            name="registerNumber"
            value={car.registerNumber}
          />
          <TextField
            variant="standard"
            placeholder="Price"
            onChange={handleChange}
            name="price"
            value={car.price}
          />
          </Stack>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose}>Cancel</Button>
          <Button onClick={handleSave}>Save</Button>
        </DialogActions>
      </Dialog>
    </div>
  );
}

export default AddCar;
