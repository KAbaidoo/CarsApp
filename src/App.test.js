import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';
import AddCar from './components/AddCar';

// Render-based snapshot using testing-library
test('renders AddCar component correctly', () => {
  const { container } = render(<AddCar />);
  // basic smoke: button should be present
  expect(screen.getByText(/Add Car/i)).toBeInTheDocument();
  // snapshot the DOM output
  expect(container.innerHTML).toMatchSnapshot();
});

// Test if main components render and dialog opens
test('open add car dialog', () => {
  render(<App />);
  const addButton = screen.getByText(/Add Car/i);
  fireEvent.click(addButton);

  const dialog = screen.getByRole('dialog');
  expect(dialog).toHaveTextContent(/Add New Car/i);
});
