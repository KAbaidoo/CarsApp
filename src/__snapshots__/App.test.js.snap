// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddCar component correctly 1`] = `"<div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\">Add Car</button></div>"`;
